src/Squisher.js

```diff
@@ -39,229 +39,59 @@ class Squisher {
+            this.updateHelper(node.children[i], squished);
         }
     }
-    
-    squish(entity) {
-        // Type (1) + Player ID (1) + Size (1) + color (4) + pos (4) + size (4) + text position (2) + text (32) + assets (37 * assetCount)
-        // TODO: store type in array to stop sending unnecessary data 
-        let squishedSize = 1 + 1 + 1 + 4 + 4 + 4 + (entity.text ? 2 + 32 : 0) + (entity.assets ? 37 * Object.keys(entity.assets).length : 0);
-
-        const squished = new Array(squishedSize);
-        let squishedIndex = 0;
-        squished[squishedIndex++] = 3;
-        squished[squishedIndex++] = entity.playerId;
-        squished[squishedIndex++] = squished.length;
- 
-        if (!(entity.pos && entity.color && entity.size)) {
-            return squished;
-        }
-        
-        squished[squishedIndex++] = entity.color[0];
-        squished[squishedIndex++] = entity.color[1];
-        squished[squishedIndex++] = entity.color[2];
-        squished[squishedIndex++] = entity.color[3];
-
-        squished[squishedIndex++] = Math.floor(entity.pos.x);
-        squished[squishedIndex++] = Math.floor(100 * (entity.pos.x - Math.floor(entity.pos.x)));
-
-        squished[squishedIndex++] = Math.floor(entity.pos.y);
-        squished[squishedIndex++] = Math.floor( 100 * (entity.pos.y - Math.floor(entity.pos.y)));
-
-        squished[squishedIndex++] = Math.floor(entity.size.x);
-        squished[squishedIndex++] = Math.floor(100 * (entity.size.x - Math.floor(entity.size.x)));
 
-        squished[squishedIndex++] = Math.floor(entity.size.y);
-        squished[squishedIndex++] = Math.floor(100 * (entity.size.y - Math.floor(entity.size.y)));
-
-        if (entity.text) {
-            squished[squishedIndex++] = entity.text && entity.text.x;
-            squished[squishedIndex++] = entity.text && entity.text.y;
-
-            let textIndex = 0;
-            while (entity.text && textIndex < 32) {
-                if (textIndex < entity.text.text.length) {
-                    squished[squishedIndex++] = entity.text.text.charCodeAt(textIndex);
-                } else {
-                    squished[squishedIndex++] = null;
-                }
-                textIndex++;
-            }
-        }
-        
-        if (entity.assets) {
-            for (const key in entity.assets) {
-                const asset = entity.assets[key];
-                squished[squishedIndex++] = asset.pos.x;
-                squished[squishedIndex++] = asset.pos.y;
-                squished[squishedIndex++] = asset.size.x;
-                squished[squishedIndex++] = asset.size.y;
-                for (let i = 0; i < 32; i++) {
-                    if (i < key.length) {
-                        squished[squishedIndex++] = key.charCodeAt(i);
-                    } else {
-                        squished[squishedIndex++] = null;
-                    }
-                }
-            }
+    handleStateChange(node) {
+        // todo: fix this
```

Comment:
this TODO will remain, since it's referencing the whole "every update results in a re-render from the root" problem. We should probably make a github issue for this and reference it whenever this issue comes up